doc/develop/test/twister_statuses.rst

```diff
@@ -0,0 +1,215 @@
+.. _twister_statuses:
+
+Twister Status
+##############
+
+What is a Twister Status?
+=========================
+
+Twister Status formulates the current state of
+
+- ``Harness``
+- ``TestCase``
+- ``TestSuite``
+- ``TestInstance``
+
+in a comprehensive and easy-to understand way.
+In practice, most users will be interested in Statuses
+of Instances and Cases after the conclusion of their Twister runs.
+
+.. tip::
+
+   Nomenclature reminder:
+
+   .. tabs::
+
+      .. tab:: ``Harness``
+
+         ``Harness`` is a Python class inside Twister that allows us
+         to capture and analyse output from a program external to Twister.
+         It has been excised from this page for clarity,
+         as it does not appear in final reports.
+
+      .. tab:: ``TestCase``
+
+         ``TestCase``, also called Case, is a piece of code that aims to verify some assertion.
+         It is the smallest subdivision of testing in Zephyr.
+
+      .. tab:: ``TestSuite``
+
+         ``TestSuite``, also called Suite, is a grouping of Cases.
+         One can modify Twister's behaviour on a per-Suite basis via ``testcase.yaml`` files.
+         Such grouped Cases should have enough in common
+         for it to make sense to treat them all the same by Twister.
+
+      .. tab:: ``TestInstance``
+
+         ``TestInstance``, also called Instance, is a Suite on some platform.
+         Twister typically reports its results for Instances,
+         despite them being called "Suites" there.
+         If a status is marked as applicable for Suites, it is also applicable for Instances.
+         As the distinction between them is not useful in this section,
+         whenever you read "Suite", assume the same for Instances.
```

Comment:
Twister summary output uses `test configurations executed on platforms` for 'instances'